Update legal.md to expand and clarify DCO option (#1127)
github/opensource.guide: _articles/legal.md

Some situations where you may want to consider an additional contributor agreement for your project include:

* Your lawyers want all contributors to expressly accept (_sign_, online or offline) contribution terms, perhaps because they feel the open source license itself is not enough (even though it is!). If this is the only concern, a contributor agreement that affirms the project's open source license should be enough. The [jQuery Individual Contributor License Agreement](https://contribute.jquery.org/CLA/) is a good example of a lightweight additional contributor agreement.

> * For some projects, all you or your lawyers may want is a simple representation that a developer was authorized to contribute their contribution. Would you want to accept a contribution from someone who could not represent they were authorized to contribute that contribution? A [Developer Certificate of Origin](https://developercertificate.org/) requirement can be a simple option. There is also a very useful tool to automate enforcement of the DCO on your repository called the [DCO Probot](https://github.com/probot/dco). One key difference with a CLA is the DCO `signed-off-by` representation is made on every commit, whereas CLAs are signed once and forgotten. This is what the [Node.js community](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md) uses today instead of their prior CLA requirement. It's also important to know many projects that have a CLA will also require a DCO so that the community can have confidence each commit was authorized.

Reviewer: @mkdolan thanks for this PR! I've made a suggested edit that I hope captures all of the key facts, but is briefer, and avoids casting aspersions on regular CLAs:

```suggestion
* You or your lawyers want developers to represent that each commit they make is authorized. A [Developer Certificate of Origin](https://developercertificate.org/) requirement is how many projects achieve this. For example, the Node.js community [uses](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md) the DCO [instead](https://nodejs.org/en/blog/uncategorized/notes-from-the-road/#easier-contribution) of their prior CLA. A simple option to automate enforcement of the DCO on your repository is the [DCO Probot](https://github.com/probot/dco).
```

Is this still accurate in your view?

I did cut the last sentence entirely, but if there is a good example of "many projects that have a CLA will also require a DCO" happy to add that fact. (I haven't heard of such projects, but would love to know of them.)

Author: Hi @mlinksva unfortunately you're mixing two different things here. A CLA is not the same as a DCO. For example, Node.js had a CLA (Joyent managed this), then they dropped the CLA entirely. When we started merging io.js (the fork) and Node.js back into one community we worked with them to start requiring the DCO. So for example saying Node.js switched from a CLA to a DCO is misleading, not just because of the timeline but the intent of each was very different. The DCO was to capture representations the developers were authorized to contribute what they're contributing under the license of the project. The CLA was to get a corporate signature from the Company that they were bound to the terms of the CLA - independent of the open source license of the project. The DCO also works WITH CLAs and in fact every project the LF hosts that has a CLA also requires the DCO on every commit. One key advantage to adding a CLA to to capture a [signature of an authorized corporate authority](https://opensource.google/docs/cla/policy/#corporate-signing-authority) binding the company to the CLA terms, which often include a patent grant. This is usually due to patent commitments in the underlying open source license of the project and wanting to add an additional layer of protection by having the company sign off. Does that make sense?

Reviewer 2: Hola

Reviewer: @mkdolan (apologies for delayed response) yes, it makes sense. Do you have a link documenting

> every project the LF hosts that has a CLA also requires the DCO on every commit

for one of these projects? It will be really helpful for readers to have a concrete example.

Reviewer: ```suggestion
* You or your lawyers want developers to represent that each commit they make is authorized. A [Developer Certificate of Origin](https://developercertificate.org/) requirement is how many projects achieve this. For example, the Node.js community [uses](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md) the DCO [instead](https://nodejs.org/en/blog/uncategorized/notes-from-the-road/#easier-contribution) of their prior CLA. A simple option to automate enforcement of the DCO on your repository is the [DCO Probot](https://github.com/probot/dco).
```

Reviewer: I'm going to go ahead and merge with my suggestion, as it's on net a clear improvement over what's in the guide now. Happy to add something about the CLA+DCO use case with supporting example if anyone would like to point one out later.
